source/ConversationPanel.cpp

```diff
@@ -351,6 +351,10 @@ bool ConversationPanel::Hover(int x, int y)
 // The player just selected the given choice.
 void ConversationPanel::Goto(int index, int selectedChoice)
 {
+	auto getter = [&](const std::string &str, size_t start, size_t length)
+	{
+		return player.Conditions().Get(str.substr(start, length));
+	};
```

Comment:
Can the caller not handle the substring? So this lambda need only retrieve the given condition string?
```suggestion
	auto getter = [&](const string &str) -> int64_t
	{
		return player.Conditions().Get(str);
	};
```
@quyykk capture everything or just `player`?

Reply:
The `Format` class has no knowledge of `ConditionsStore`. It should provide the form that allows as much freedom as possible to the caller, in case further optimization is possible.

Reply:
> @quyykk capture everything or just player?

No, GCC won't let me do that. It wants `this` if I'm passing a non-static data member.

Reply:
I'd also rather not assume the return type of `player.Conditions.Get()`, just in case it is upgraded one day.

Reply:
The getter no longer passes everything. Instead, it passes `player.Conditions()` in a `conditions` variable.